requests_ntlm/requests_ntlm.py

```diff
@@ -131,7 +131,7 @@ def retry_using_http_NTLM_auth(self, auth_header_field, auth_header,
         response3.history.append(response2)
 
         # Get the session_security object created by ntlm-auth for signing and sealing of messages
-        self.session_security = context.session_security
+        self.session_security = context._session_security
```

Comment:
This one is a tricky one, technically the `wrap` and `unwrap` methods of `NtlmContext()` should be used instead but unfortunately `unwrap` on `NtlmContext` has a different signature than the one of session_security. While this works I don't really want to rely on `_` prefixed attributes because they aren't really guaranteed to stay stable over different releases.

What I recommend is to add a session_security property to this class like so:

```
@property
def session_security(self):
    warnings.warn("Deprecated property session_security, the wrap and unwrap methods should be accessed using the host context ntlm.contexts['hostname'].wrap/unwrap instead.", DeprecationWarning)
    return next(iter(self.contexts))._session_security
```

This also requires us to store the `NtlmContext` as a property in requests-ntlm. This is done in a similar way with requests-kerberos where the context is stored in a dict that's indexed with the hostname of the request. You can see that [requests-kerberos](https://github.com/requests/requests-kerberos/blob/master/requests_kerberos/kerberos_.py) uses urlparse `from requests.compat import urlparse` to get the hostname from the request and that is passed into the retry method. From there the context that is initialized is stored in `self.contexts[hostname]`.

Doing this means that other libraries that rely on the wrapping and unwrapping functions provided by session_security aren't broken in the next release, are warned about it being deprecated here and are offered the alternative.